packages/nimbus/src/components/data-table/components/data-table.row.tsx

```diff
@@ -0,0 +1,289 @@
+import { forwardRef, useState } from "react";
+import { Row as RaRow, Cell as RaCell } from "react-aria-components";
+import { Highlight } from "@chakra-ui/react";
+import { useDataTableContext } from "./data-table.root";
+import { DataTableExpandButton } from "../data-table.slots";
+import type { DataTableRow as DataTableRowType } from "../data-table.types";
+import { Box, Checkbox, Button, IconButton } from "@/components";
+import { useCopyToClipboard } from "@/hooks";
+import {
+  KeyboardArrowDown,
+  KeyboardArrowRight,
+  ContentCopy,
+  Info,
+} from "@commercetools/nimbus-icons";
+
+export interface DataTableRowProps {
+  row: DataTableRowType<any>;
+  depth?: number;
+}
+
+export const DataTableRow = forwardRef<HTMLTableRowElement, DataTableRowProps>(
+  function DataTableRow({ row, depth = 0 }, ref) {
+    const {
+      visibleCols,
+      search,
+      expanded,
+      toggleExpand,
+      nestedKey,
+      showExpandColumn,
+      showSelectionColumn,
+      showDetailsColumn,
+      isRowClickable,
+      isTruncated,
+      onRowClick,
+      onDetailsClick,
+      isSelected,
+      handleRowSelection,
+      isDisabled,
+      onRowAction,
+    } = useDataTableContext();
+
+    // Hover state management - only for copy functionality
+    const [hoveredCell, setHoveredCell] = useState<string | null>(null);
+    const [, copyToClipboard] = useCopyToClipboard();
+
+    const hasNestedContent =
+      nestedKey &&
+      row[nestedKey] &&
+      (Array.isArray(row[nestedKey]) ? row[nestedKey].length > 0 : true);
+    const isExpanded = expanded[row.id];
+
+    const handleRowClick = () => {
+      if (isDisabled(row.id)) {
+        if (onRowAction) {
+          onRowAction(row, "click");
+        }
+        return;
+      }
+
+      if (isRowClickable && onRowClick) {
+        onRowClick(row);
+      }
+    };
+
+    // Action handlers - only copy functionality
+    const handleCopy = (value: any) => {
+      const textValue = typeof value === "string" ? value : String(value);
+      copyToClipboard(textValue);
+    };
+
+    // Highlight helper
+    const highlightCell = (value: any) =>
+      search && typeof value === "string" ? (
+        <Highlight query={search}>{value}</Highlight>
+      ) : (
+        value
+      );
+
+    return (
+      <>
+        <RaRow
```

Comment:
A lot of the checking happening for whether a row is disabled that's happening here is handled by react-aria if the `isDisabled` prop is set.
```suggestion
        <RaRow
          isDisabled={isDisabled}
```